### ISSUE TITLE
[Solutions] 14th

### ISSUE THREAD

**Issue by riow1983** [riow1983/Kaggle-Coleridge-Initiative#20]
[Discussion]
[14th Place Solution (with notebooks)](https://www.kaggle.com/c/coleridgeinitiative-show-us-the-data/discussion/248511)

[Code]
- Additional hand-labeled dataset titles (from train only): https://www.kaggle.com/lichena/coleridgehandlabeled
- Preprocessing: https://www.kaggle.com/lichena/coleridge-pre-processing-hand
- Training: https://www.kaggle.com/lichena/coleridge-ner?scriptVersionId=66238738
- Inference: https://www.kaggle.com/lichena/coleridge-ner-inference

[一言で言うと]
trainから目視で抜いてきた"hand label"を使った, fine-tuned bert-base-cased + crf on BIO-tagged chunks of ~200-400 words
<br>
<br>

***
[関連技術]
- PyTorch CRF
[pytorch-crf](https://pytorch-crf.readthedocs.io/en/stable/)

**Comment by riow1983** [comment]
[ラベル候補の取得]
正規表現の検索結果をベースに, 手動アノテーションでラベル候補を935個用意した.
(学習データセットにある教師ラベル数は130個だったので7倍以上の増幅.)

[FP除外]
FPリストを作成し, ラベル候補を選抜.

[データセット言及があるセンテンスで学習]
データセット言及があるセンテンス(全体の80%)でtrain, 言及がないセンテンス(全体の20%)でvalid.
(データセット言及があるセンテンスの20%でCVを作成してみたが, private LBに対して汎化しなかった.)

使用モデル: fine-tuned bert-base-cased + crf on BIO-tagged chunks of ~200-400 words


その他Tips:
大文字小文字の区別, 括弧類は残して学習させたほうが精度が上がった.

**Comment by riow1983** [comment]
スライド作成完了.
